### PR TITLE
fix(schema-to-ts): Simplify Array types (IQQ-1490)

### PR DESCRIPTION
**Pull request by tkcranny**
This removes length-enforced tuples types with spread parameters that are difficult and annoying to write in separate declarations.

## Example:

```typescript
const choices: [FieldChoiceWithLabel, ...FieldChoiceWithLabel[] = [...]
```

becomes just:

```typescript
const choices: FieldChoiceWithLabel[] = [...]
```

cc @cvanderkolk

### CODE REVIEW

**Review by eliangcs — APPROVED** [review]

File: packages/core/types/zapier.generated.d.ts [hunk 1]
```diff
@@ -290,10 +290,8 @@ export interface Field {
    * sub-object for this field. Usually used for line items.
    *
    * @minItems 1
-   *
-   * @minItems 1
    */
-  children?: [Field, ...Field[]];
+  children?: Field[];
```
Comment (szchenghuang) [inline packages/core/types/zapier.generated.d.ts:294]: Nice optimization!

File: packages/core/types/zapier.generated.d.ts [hunk 2]
```diff
@@ -185,7 +185,7 @@ export type RefResource = string;
  */
 export type FieldChoices =
   | { [k: string]: unknown }
```
Comment (szchenghuang) [inline packages/core/types/zapier.generated.d.ts:187]: I know this line is out of scope. Just thinking out loud here. It can be written as `Record<string, unknown>` alternatively.
